lib/src/native/realm_core.dart

```diff
@@ -326,6 +326,47 @@ class _RealmCore {
     return RealmResultsHandle._(pointer);
   }
 
+  RealmResultsHandle queryTable(Realm realm, int classKey, String query, List<Object> args) {
+    return using((arena) {
+      final length = args.length;
+      final argsPointer = arena<Pointer<realm_value_t>>(length);
+      for (var i = 0; i < length; ++i) {
+        argsPointer[0] = _toRealmValue(args[i], arena);
+      }
+      final queryHandle = RealmQueryHandle._(_realmLib.invokeGetPointer(
+        () => _realmLib.realm_query_parse(
+          realm.handle._pointer,
+          classKey,
+          query.toUtf8Ptr<Int8>(arena),
+          length,
+          argsPointer[0],
+        ),
+      ));
+      final resultsPointer = _realmLib.invokeGetPointer(() => _realmLib.realm_query_find_all(queryHandle._pointer));
+      return RealmResultsHandle._(resultsPointer);
+    });
+  }
+
+  RealmResultsHandle queryResults(Realm realm, RealmResults target, String query, List<Object> args) {
+    return using((arena) {
+      final length = args.length;
+      final argsPointer = arena<Pointer<realm_value_t>>(length);
+      for (var i = 0; i < length; ++i) {
+        argsPointer[0] = _toRealmValue(args[i], arena);
```

Comment:
this sets only the first arg

Reply:
Doh! Good catch. Will add more thorough test

Reply:
Fixed